packages/vector_graphics_compiler/lib/src/svg/visitor.dart

```diff
@@ -0,0 +1,317 @@
+import '../draw_command_builder.dart';
+import '../geometry/basic_types.dart';
+import '../geometry/matrix.dart';
+import '../geometry/path.dart';
+import '../paint.dart';
+import '../vector_instructions.dart';
+import 'node.dart';
+import 'parser.dart';
+
+/// A visitor implementation used to process the tree.
```

Comment:
We should document what `S` is used for, and probably just make all of these `void` instead of generic if we can.

Reply:
We absolutely use non-void return types here, please take a closer look at how resolving visitor is implemented

Reply:
It _could_ be implemented by having a member variable, but it's ok.

My hackles went up here because this pattern in the Dart kernel package ends up being used with a dynamic return type and it's super hard to figure out what's going on. I'm ok as long as we don't do that.